common/src/main/java/tachyon/worker/WorkerClient.java

```diff
@@ -325,8 +312,12 @@ public synchronized String lockBlock(long blockId) throws IOException {
 
     try {
       return mClient.lockBlock(blockId, mSessionId);
-    } catch (FileDoesNotExistException e) {
-      return null;
+    } catch (TachyonTException e) {
+      if (e.getType().equals(TachyonExceptionType.FILE_DOES_NOT_EXIST.name())) {
```

Comment:
The point of this is to minimize the amount of code that needs to understand the Thrift exceptions.